src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -65,10 +67,9 @@ public static IServiceCollection AddAsyncInitializer<TInitializer>(this IService
         /// <returns>A reference to this instance after the operation has completed.</returns>
         public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Func<IServiceProvider, IAsyncInitializer> implementationFactory)
         {
-            if (implementationFactory == null)
-                throw new ArgumentNullException(nameof(implementationFactory));
-
-            return services
+            return implementationFactory == null
+                ? throw new ArgumentNullException(nameof(implementationFactory))
+                : services
```

Comment:
Please undo this change, it makes the code less readable.
(and it's not related to the feature you're implementing anyway)